### ISSUE TITLE
Can it colorize big resolution images?

### ISSUE THREAD

**Issue by FurkanGozukara** [ErwannMillon/Color-diffusion#5]
i plan to put some time into making a tutorial for this

but before spending time i want to learn this

it is

    parser.add_argument("--image-size", default=64, dest="img_size", type=int)

so can we colorize like 1024x1024 images as well or only 64px?

i am not interested in training

@ErwannMillon

**Comment by saulchiu** [comment]
> i plan to put some time into making a tutorial for this
> 
> but before spending time i want to learn this
> 
> it is
> 
> ```
> parser.add_argument("--image-size", default=64, dest="img_size", type=int)
> ```
> 
> so can we colorize like 1024x1024 images as well or only 64px?
> 
> i am not interested in training
> 
> [@ErwannMillon](https://github.com/ErwannMillon)

Hi, sorry to bother you, have you solve this problem now? Recently I face the same problem too: when training for colorization for dataset with 256*256 resolusion, the loss is hard to decrease.

**Comment by XTY-starstar** [comment]
> > 我打算花一些时间为此制作一个教程
> > but before spending time i want to learn this
> > it is
> > ```
> > parser.add_argument("--image-size", default=64, dest="img_size", type=int)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > so can we colorize like 1024x1024 images as well or only 64px?
> > i am not interested in training
> > [@ErwannMillon](https://github.com/ErwannMillon)
> 
> Hi, sorry to bother you, have you solve this problem now? Recently I face the same problem too: when training for colorization for dataset with 256*256 resolusion, the loss is hard to decrease.

Hello bro, sorry to bother you, have you solve this problem now? Recently I face the same problem tooooooo

**Comment by saulchiu** [comment]
> > > 我打算花一些时间为此制作一个教程
> > > but before spending time i want to learn this
> > > it is
> > > ```
> > > parser.add_argument("--image-size", default=64, dest="img_size", type=int)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > so can we colorize like 1024x1024 images as well or only 64px?
> > > i am not interested in training
> > > [@ErwannMillon](https://github.com/ErwannMillon)
> > 
> > 
> > Hi, sorry to bother you, have you solve this problem now? Recently I face the same problem too: when training for colorization for dataset with 256*256 resolusion, the loss is hard to decrease.
> 
> Hello bro, sorry to bother you, have you solve this problem now? Recently I face the same problem tooooooo

nope, i change another way to do colorization. Here is the process:
```python
    x_lab = rgb_tensor_to_lab_tensor(x_rgb)
    x_l, _ = split_lab_channels(x_lab)
    l_rgb = x_l.repeat(1, 3, 1, 1).to(x_rgb)
    x_t = l_rgb.clone()
    col_list.append(l_rgb.cpu().clone())

    distance = int(t / loop)
    decreasing_list = [t - i * distance for i in range(loop)]
    print(decreasing_list)
    for i in tqdm(range(0, loop)):
        x_t = diffusion.q_sample(x_t, (torch.ones(size=(l_rgb.shape[0],), device=device) * decreasing_list[i]).to(torch.int64))
        for j in reversed(range(0, decreasing_list[i])):
            x_t_m_1 = diffusion.p_sample(x_t, torch.tensor([j], device=device))
            x_t = x_t_m_1
        x_t = rgb_tensor_to_lab_tensor(x_t)
        _, x_ab = split_lab_channels(x_t)
        x_t = torch.cat((x_l, x_ab), dim=1)
        x_t = lab_tensor_to_rgb_tensor(x_t).to(device)
        col_list.append(x_t.cpu().clone())
```
and the visulization:
original ---- gray ---- colorized
![Image](https://github.com/user-attachments/assets/121199a5-9683-4ac7-a89c-76661487c7a3)
hope it can help u!

**Comment by XTY-starstar** [comment]
> > Hello bro, sorry to bother you, have you solve this problem now? Recently I face the same problem tooooooo
> 
> nope, i change another way to do colorization. Here is the process:
> 
>     x_lab = rgb_tensor_to_lab_tensor(x_rgb)
>     x_l, _ = split_lab_channels(x_lab)
>     l_rgb = x_l.repeat(1, 3, 1, 1).to(x_rgb)
>     x_t = l_rgb.clone()
>     col_list.append(l_rgb.cpu().clone())
> 
>     distance = int(t / loop)
>     decreasing_list = [t - i * distance for i in range(loop)]
>     print(decreasing_list)
>     for i in tqdm(range(0, loop)):
>         x_t = diffusion.q_sample(x_t, (torch.ones(size=(l_rgb.shape[0],), device=device) * decreasing_list[i]).to(torch.int64))
>         for j in reversed(range(0, decreasing_list[i])):
>             x_t_m_1 = diffusion.p_sample(x_t, torch.tensor([j], device=device))
>             x_t = x_t_m_1
>         x_t = rgb_tensor_to_lab_tensor(x_t)
>         _, x_ab = split_lab_channels(x_t)
>         x_t = torch.cat((x_l, x_ab), dim=1)
>         x_t = lab_tensor_to_rgb_tensor(x_t).to(device)
>         col_list.append(x_t.cpu().clone())
> and the visulization: original ---- gray ---- colorized ![Image](https://github.com/user-attachments/assets/121199a5-9683-4ac7-a89c-76661487c7a3) hope it can help u!


Impressive work! Could you please provide a more detailed version of the code, including q_sample、p_sample and so  on.
@chengyiqiu1121

**Comment by saulchiu** [comment]
> > > Hello bro, sorry to bother you, have you solve this problem now? Recently I face the same problem tooooooo
> > 
> > 
> > nope, i change another way to do colorization. Here is the process:
> > ```
> > x_lab = rgb_tensor_to_lab_tensor(x_rgb)
> > x_l, _ = split_lab_channels(x_lab)
> > l_rgb = x_l.repeat(1, 3, 1, 1).to(x_rgb)
> > x_t = l_rgb.clone()
> > col_list.append(l_rgb.cpu().clone())
> > 
> > distance = int(t / loop)
> > decreasing_list = [t - i * distance for i in range(loop)]
> > print(decreasing_list)
> > for i in tqdm(range(0, loop)):
> >     x_t = diffusion.q_sample(x_t, (torch.ones(size=(l_rgb.shape[0],), device=device) * decreasing_list[i]).to(torch.int64))
> >     for j in reversed(range(0, decreasing_list[i])):
> >         x_t_m_1 = diffusion.p_sample(x_t, torch.tensor([j], device=device))
> >         x_t = x_t_m_1
> >     x_t = rgb_tensor_to_lab_tensor(x_t)
> >     _, x_ab = split_lab_channels(x_t)
> >     x_t = torch.cat((x_l, x_ab), dim=1)
> >     x_t = lab_tensor_to_rgb_tensor(x_t).to(device)
> >     col_list.append(x_t.cpu().clone())
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and the visulization: original ---- gray ---- colorized ![Image](https://github.com/user-attachments/assets/121199a5-9683-4ac7-a89c-76661487c7a3) hope it can help u!
> 
> Impressive work! Could you please provide a more detailed version of the code, including q_sample、p_sample and so on. [@chengyiqiu1121](https://github.com/chengyiqiu1121)
Thanks! The APIs (q_sample and p_sample...) are the same as this opensource [repository](https://github.com/lucidrains/denoising-diffusion-pytorch), you can refer to this repo to get more informations. Besides, i will open my repository in the future.
